### ISSUE TITLE
html parser ignores "display: inline" in some cases

### ISSUE THREAD

**Issue by lazka** [prettier/prettier#12218]
**Prettier 2.5.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAEQPIGEAqAmgAoCiABABYwC2ANgHwA6Uq19zU535qAJgEsAbgwBmECBgC0U8oIDOABzoBDAJ5JyAqHW1xyMhqiUqoDAEYqATqgD0Js3cEiWd9oxYgANCAiKYAtDyyKDWVhAA7sTWCMEoKnQR6sE+5lYqYADWcDAAyio0cAAyesiiCfJwqelZObmKGdoA5sgwVgCuVSBwNOZwfHz9RaZN7SpNcABiEFY0KjABUC3x7TAQ3iDuAOqUAvAmYHC5sXvCe2rIIAJNUDNwG9qVVjDE6U1zZRVdAFbyAB65Zp0OAARXaEHgnzolR8DSsT0u7g2iis2hgWwEfBglGQAA4AAyw8KVLbpRSXFFwJ5Ce4+ACO4Pgrz8cRAKnkUigcH6-Q2VjgDIE-Ne4w+SHK0K6lRoAlaHSlQNBjPu4q+PhgKnMGKxOKQACZ1ekBLolrgIDQxd15ABWDbtSr4TVxCUwkBCToASSgg1guTAqP8AEFvbkYGpgVDKgBfKNAA)

```sh
--parser html
--html-whitespace-sensitivity ignore
```

**Input:**

```html
<!DOCTYPE html>
<html>
    <div>foo<!-- display: inline --><span>bar</span></div>
</html>

```

**Output:**

```html
<!DOCTYPE html>
<html>
  <div>
    foo
    <!-- display: inline -->
    <span>bar</span>
  </div>
</html>

```

**Expected behavior:**

Since I set the span element to display inline I expected prettier to preserve the whitespace

```html
<!DOCTYPE html>
<html>
  <div>
    foo<!-- display: inline --><span>bar</span>
  </div>
</html>

```

**Comment by duailibe** [comment]
Hi @lazka 

Since you've set `--html-whitespace-sensitivity ignore`, all whitespace is considered insignificant, so Prettier can ignore any magic comments and insert whitespace wherever it wants.

If you change it to `css`, it works like you probably expect:

**Prettier 2.5.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAEQPIGEAqAmgAoCiABABYwC2ANgHwA6Uq19zU535qAJgEsAbgwBmECBgC0U8oIDOABzoBDAJ5JyAqHW1xyMhqiUqoDAEYqATqgD0Js3cEiWd9oxAAaEBEUwB0PLIoNZWEADuxNYIQSgqdOHqQd7mVipgANZwMADKKjRwADJ6yKLx8nApaZnZOYrp2gDmyDBWAK6VIHA05nB8fH2Fpo1tKo1wAGIQVjQqMP5QzXFtMBBeIO4A6pQC8CZgcDkxu8K7asjg8skg2hVWMMRpjbOl5Z0AVvIAHjlNdHAARTaEHgrzoFW89Ssdwu7nWiis2hgmwEfBglGQAA4AAyQsIVTZpRQXBFwO5CODrACOwPgj18sRAKnkUigcD6fXWVjgNIE3MeYxeSDK4M6FRoAha7TFf0BtMpwre3hgKnMKLRGKQACZlWkBLpFrgIDQhV15ABWdZtCr4VWxEUQkBCDoASSgA1gOTAiL8AEF3TkYGp-mCKgBfMNAA)

```sh
--parser html
```

**Input:**

```html
<!DOCTYPE html>
<html>
    <div>foo<!-- display: inline --><span>bar</span></div>
</html>
```

**Output:**

```html
<!DOCTYPE html>
<html>
  <div>foo<!-- display: inline --><span>bar</span></div>
</html>

```

**Comment by lazka** [comment]
> Since you've set `--html-whitespace-sensitivity ignore`, all whitespace is considered insignificant, so Prettier can ignore any magic comments and insert whitespace wherever it wants.

I don't think that's true, this works just fine: [Playground Link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAEQPIGEAqAmgAoCiABABYwC2ANgHwA6Uq19zU535GAtH3IATAJYBnAA50AhgE8k5EVDpK45AZx69RANwYAzCBFSTpUBgCNpAJ1QB6U+fu7O99oxYgANCAgSYItBiyKA21hAA7sQ2CMEo0nQRcsE+FtbSYADWcDAAytI0cAAyqsj6CWJwqelZObkSGUoA5sgw1gCuVSBwNBZwQkL9RWZN7dJNcABiENY00jABUC3x7TAQ3iDuAOqUIvCmYHC5sXsiOnuyyCAiTVAzcBtKldYwxOlNc2UVXQBWYgAeuWadDgAEV2hB4F86JUfA1rM8ru4NhJrEoYFsREIYJRkAAOAAMcPClS26QkV1RcGeOgePgAjhD4G8-HEQNIxHwoHB+v0NtY4IyRAK3uNPkhyjCupUaCJWh1pcCwUyHhLvj4YNILJjsbikAAmDXpEQqJa4CA0cXdMQAVg27Uq+C1cUlsJAOk6AEkoINYLkwGj-ABBH25GCyEHQyoAX2jQA)

The `display: inline` makes the diff respect whitespace despite html-whitespace-sensitivity=ignore.

**Comment by lazka** [comment]
Yes, the problem is that whitespace surrounding the (inline) span element should be preserved, but isn't. And it breaks the rendered HTML as a result.

**Comment by duailibe** [comment]
If you want the whitespace surround the span to be preserved, you need to mark the parent as inline.

`foo<!-- display: inline --><span>bar</span>` means that `<span>bar</span>` should be printed as inline (and it is printed as inline).

I don't understand what is wrong yet.

**Comment by duailibe** [comment]
@lazka See https://github.com/prettier/prettier/pull/12219

Can you test it (and see if there are any regressions) with some real examples?
